rules_daml/daml.bzl

```diff
@@ -1,94 +1,299 @@
 # Copyright (c) 2020 The DAML Authors. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-load("@bazel_skylib//lib:paths.bzl", "paths")
-load("@build_environment//:configuration.bzl", "ghc_version")
-
-daml_provider = provider(doc = "DAML provider", fields = {
-    "dalf": "The DAML-LF file.",
-    "dar": "The packaged archive.",
-})
-
-def _daml_impl_compile_dalf(ctx):
-    # Call damlc compile
-    compile_args = ctx.actions.args()
-    compile_args.add("compile")
-    compile_args.add(ctx.file.main_src)
-    compile_args.add("--output", ctx.outputs.dalf.path)
-    if ctx.attr.target:
-        compile_args.add("--target", ctx.attr.target)
-    ctx.actions.run(
-        inputs = depset([ctx.file.main_src] + ctx.files.srcs),
-        outputs = [ctx.outputs.dalf],
-        arguments = [compile_args],
-        progress_message = "Compiling DAML into DAML-LF archive %s" % ctx.outputs.dalf.short_path,
-        executable = ctx.executable.damlc,
-    )
-
-def _daml_impl_package_dar(ctx):
-    # Call damlc package
-    package_args = ctx.actions.args()
-    package_args.add("package")
-    package_args.add(ctx.file.main_src)
-    package_args.add(ctx.attr.name)
-    if ctx.attr.target:
-        package_args.add("--target", ctx.attr.target)
-    package_args.add("--output")
-    package_args.add(ctx.outputs.dar.path)
-    ctx.actions.run(
-        inputs = [ctx.file.main_src] + ctx.files.srcs,
-        outputs = [ctx.outputs.dar],
-        arguments = [package_args],
-        progress_message = "Creating DAR package %s" % ctx.outputs.dar.basename,
-        executable = ctx.executable.damlc,
-    )
-
-def _daml_compile_impl(ctx):
-    _daml_impl_compile_dalf(ctx)
-    _daml_impl_package_dar(ctx)
-
-    # DAML provider
-    daml = daml_provider(
-        dalf = ctx.outputs.dalf,
-        dar = ctx.outputs.dar,
-    )
-    return [daml]
-
-def _daml_compile_outputs_impl(name):
-    patterns = {
-        "dalf": "{name}.dalf",
-        "dar": "{name}.dar",
-    }
-    return {
-        k: v.format(name = name)
-        for (k, v) in patterns.items()
-    }
-
-daml_compile = rule(
-    implementation = _daml_compile_impl,
+load("@build_environment//:configuration.bzl", "ghc_version", "sdk_version")
+
+_damlc = attr.label(
+    allow_single_file = True,
+    default = Label("//compiler/damlc"),
+    executable = True,
+    cfg = "host",
+    doc = "The DAML compiler.",
+)
+
+_zipper = attr.label(
+    allow_single_file = True,
+    default = Label("@bazel_tools//tools/zip:zipper"),
+    executable = True,
+    cfg = "host",
+)
+
+def _daml_configure_impl(ctx):
+    project_name = ctx.attr.project_name
+    project_version = ctx.attr.project_version
+    daml_yaml = ctx.outputs.daml_yaml
+    target = ctx.attr.target
+    ctx.actions.write(
+        output = daml_yaml,
+        content = """
+            sdk-version: {sdk}
```

Comment:
This needs to be `ghc_version` not SDK version. `sdk_version` will break when you are on a snapshot release commit.

Reply:
Thanks. @garyverhaegen-da Is there doc to explain what versions shoul dbe used where?
I made the fix in PR #5094

Reply:
`ghc_version` needs to be used for anything that ends up as the version of a GHC (DAML) package, e.g. for the version of `daml-stdlib`. `sdk-version` is the “official” way to refer to an SDK version e.g. in the assistant. In this case `sdk_version` is completely correct. Sorry for forgetting to send the comment. I keep getting confused by githubs UX.